### PR TITLE
Created automated tests using github actions. Workflow installs and runs build and unit tests. Also runs prettier formatting on both frontend and backend.

### PR DESCRIPTION
**Pull request by DaleOrders**

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          node-version: '18.16.0'
+      - name: Install backend dependencies
+        working-directory: ./server
+        run: npm ci
+      - name: Run backend build
+        working-directory: ./server
+        run: npm run build
+      - name: Run backend prettier tests
+        working-directory: ./server
+        run: npm run format
```
Comment (0916dhkim) [inline .github/workflows/node.js.yml:32]: This command is for fixing the formatting. Do we have a command for checking the formatting (but not writing to the files)?

Same for /web

**Review by 0916dhkim — COMMENTED** [review]
Looks good overall! Just one comment about the format command in the workflow.

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (DaleOrders) [reply .github/workflows/node.js.yml:32]: So you just want it to check, but not fix the files. All good. I'll do that. Thanks

Comment (DaleOrders) [reply .github/workflows/node.js.yml:32]: Ok done. Changed it to only run checks, but not format the files. Thanks

**Review by 0916dhkim — APPROVED** [review]
LGTM! :+1:

**Pull request merged**